File: 202204/excel_sheet_column_number.py
char_list = ["A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M", "N", "O", "P", "Q", "R", "S", "T",
             "U", "V", "W", "X", "Y", "Z"]

class Solution:
    def titleToNumber(self, columnTitle: str) -> int:
        result: int = 0
        for i, c in enumerate(reversed(columnTitle)):
            result += (26**i) * (char_list.index(c) + 1)
        return result


if __name__ == "__main__":
    print(Solution().titleToNumber("A"))

File: 202204/js_solution/excel_sheet_column_number.js
let char_list =  ["A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M", "N", "O", "P", "Q", "R", "S", "T",
"U", "V", "W", "X", "Y", "Z"]

var titleToNumber = function(columnTitle) {
    var result = 0
    var index_ = 0
    for(let c of columnTitle.split("").reverse().join("")) {
        result += (26 ** index_) * (char_list.indexOf(c) + 1)
        index_ ++;
    }
    return result
};


var sample = 'AB'
titleToNumber(sample);